src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/SingleEdgeLabelMappingToURIsImpl.java

```diff
@@ -0,0 +1,37 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+
+public class SingleEdgeLabelMappingToURIsImpl implements EdgeLabelMapping {
+
+    protected final String label;
+    protected final String iri;
+
+    public SingleEdgeLabelMappingToURIsImpl(final String label, final String iri){
+        this.label=label;
+        this.iri=iri;
+    }
+
+    @Override
+    public Node map(final String label) {
+        if (label.equals(this.label)) {
+            return NodeFactory.createURI(iri);
```

Comment:
Create this Node object already in the constructor. That way, it will not be created every time this function is called.